tests/src/Context/ScreenshotContext.php

```diff
@@ -17,6 +17,44 @@
  */
 class ScreenshotContext extends RawMinkContext {
 
+  /**
+   * The directory where the screenshots are saved.
+   *
+   * @var string
+   */
+  protected $screenshotsDir;
+
+  /**
+   * The location where to upload the screenshots as an Amazon S3 bucket URI.
+   *
+   * @var string
+   */
+  protected $artifactsS3Uri;
+
+  /**
+   * Constructs a new ScreenshotContext context.
+   *
+   * @param string $screenshots_dir
+   *   The directory where the screenshots are saved. The value is
+   *   passed in behat.yml.
+   * @param string $artifacts_s3_uri
+   *   An Amazon S3 bucket URI, such as s3://<bucket name>/path/to/artifacts,
+   *   where to store the screenshots. If an empty value is received, no AWS S3
+   *   upload will occur.
+   *
+   * @see tests/behat.yml.dist
+   */
+  public function __construct($screenshots_dir, $artifacts_s3_uri) {
+    $screenshots_dir = trim($screenshots_dir);
+    // If a directory has been passed, ensure the directory exists.
+    if (!empty($screenshots_dir) && !is_dir($screenshots_dir)) {
+      $screenshots_dir = rtrim($screenshots_dir, '/');
+      @mkdir($screenshots_dir, 0777, TRUE);
```

Comment:
`mkdir()` might return `FALSE` if the creation of the folder failed but the return value is not checked.